### PR TITLE
Adding eups

### PR DESCRIPTION
**Pull request by gcomoretto**
A version manager tracking product dependencies. Recipe contributed by Rubin Observatory Data Management, as we use it extensively.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/eups/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+requirements:
+  build:
+    - make
+
+  host:
+    - python >2.6
+
+  run:
+    - python >2.6
```
Comment (beckermr) [inline recipes/eups/meta.yaml:24]: Please remove these python constraints.

**Comment by ktlim** [comment]
I agree to be listed as a maintainer.

File: recipes/eups/activate.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Bootstrap EUPS
+EUPS_DIR="${CONDA_PREFIX}/eups"
+source "${EUPS_DIR}/bin/setups.sh"
```
Comment (beckermr) [inline recipes/eups/activate.sh:3]: This is going to stomp on any existing eups installs that are active. So for example, if a user has a global eups distrribution installed outside of conda, this command will mess that up. You need to backup any env vars that setups.sh is going to override.

File: recipes/eups/deactivate.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# unsetup any products to keep env clean
```
Comment (beckermr) [inline recipes/eups/deactivate.sh:1]: I'd appreciate credit for my existing work here: https://github.com/beckermr/stackvana-core/blob/master/recipe/stackvana_deactivate.sh

File: recipes/eups/deactivate.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+REMOVE=":${EUPS_DIR}/bin:"
+WORK=${WORK//$REMOVE/:}
+WORK=${WORK%:}
+WORK=${WORK#:}
+export PATH=$WORK
+
+# remove EUPS vars
+for var in EUPS_PATH EUPS_SHELL SETUP_EUPS EUPS_DIR EUPS_PKGROOT; do
+    unset $var
+done
```
Comment (beckermr) [inline recipes/eups/deactivate.sh:23]: Again you need to make sure not to step on user values of these. You have to restore the old ones.

File: recipes/eups/activate.sh (hunk 2 above)
Comment (beckermr) [reply recipes/eups/activate.sh:3]: You probably need to backup any other envvars eups might touch too. This could include LD_LIBRARY_PATH, DYLD_LIBRARY_PATH on osx, etc.

**Comment by beckermr** [comment]
A few more comments here for your consideration (I had to refresh my memory after reading my own recipes on this).

1. You probably should turn off locking by default (https://github.com/beckermr/stackvana-core/blob/master/recipe/build_impl.sh#L168) - I discussed this with Robert a lot and basically, this never really worked well and more than one process might be using the files as multiple activations of the conda env can happen.

2. I had trouble with previous version of eups using really long paths that choke on some platforms. I put in this block of code which helped a lot: https://github.com/beckermr/stackvana-core/blob/master/recipe/build_impl.sh#L172. You don't seem to be hitting that in the CI, but as the paths of the envs get longer, this will be more of an issue.

File: recipes/eups/build.sh [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+mkdir -p ${EUPS_HOME}
+
+# Install EUPS
+echo -e "\nInstalling EUPS..."
+echo "Using python at ${EUPS_PYTHON} to install EUPS"
+# echo "Configured EUPS_PKGROOT: ${EUPS_PKGROOT}"
+
+mkdir -p "${EUPS_PATH}"/{site,ups_db}
+touch "${EUPS_PATH}/ups_db/.conda_keep"
```
Comment (beckermr) [inline recipes/eups/build.sh:17]: You might need to touch the site dir here too. I forget.

File: recipes/eups/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  home: https://github.com/RobertLuptonTheGood/eups
+  license: GPL-2.0-or-later
+  license_family: GPL
+  license_file: gpl.txt
+  summary: 'A version manager tracking product dependencies'
+
+extra:
+  recipe-maintainers:
+    - gcomoretto
+    - brianv0
```
Comment (beckermr) [inline recipes/eups/meta.yaml:36]: cc @brianv0

File: recipes/eups/activate.sh (hunk 2 above)
Comment (beckermr) [reply recipes/eups/activate.sh:3]: One more thing here. The activate and deactivate scripts are called in lexicographical order. This means this script will be called after clang but before any gnu compilers. This is something to think about. You might want to inspect the compiler activation packages to figure out what you might be stepping on. Generally stepping on flags from them is a no-no as they have to be intact for the system to work. 

As an aside, I got lucky when I did this because `s` comes after `c` and `g`! ;p

Comment (ktlim) [reply recipes/eups/activate.sh:3]: Unfortunately, eups is general enough that it can modify any arbitrary environment variable.  Should we attempt to preserve and restore the entire _status quo ante_ environment (e.g. in a special file)?

Comment (beckermr) [reply recipes/eups/activate.sh:3]: Ah I was confused between eups and eups distrib. Let’s just preserve the env vars explicitly uses by eups itself. We can punt the preservation of env vars related to specific packages to those packages.

File: recipes/eups/build.sh (hunk 5 above)
Comment (gcomoretto) [reply recipes/eups/build.sh:17]: There should be no need to touch the site dir since it is populated during the build.

File: recipes/eups/meta.yaml (hunk 6 above)
Comment (brianv0) [reply recipes/eups/meta.yaml:36]: checking in, happy to help

**Comment by beckermr** [comment]
@conda-forge-admin relint

**Comment by gcomoretto** [comment]
Please let me know if there is anything else to fix.

File: recipes/eups/deactivate.sh [hunk 7]
```diff
@@ -16,10 +25,34 @@ WORK=${WORK//$REMOVE/:}
...
 unset -f unsetup
+if [[ "$BACKUP_unsetup" ]]; then
+  eval "$BACKUP_unsetup"
+  unset BACKUP_unsetup
+fi
+
+
+# restoring exisisting python path
+export PYTHONPATH=${BACKUP_PYTHONPATH}
+unset BACKUP_PYTHONPATH
```
Comment (beckermr) [inline recipes/eups/deactivate.sh:57]: I’d use a more specific prefix than backup. Something unique

Comment (gcomoretto) [reply recipes/eups/deactivate.sh:57]: Jost for PYTHONPATH or for all the backup variables?

Comment (beckermr) [reply recipes/eups/deactivate.sh:57]: All of them. Maybe CONDA_EUPS_BACKUP or somethjng like that

**Comment by beckermr** [comment]
@conda-forge-admin relint

File: recipes/eups/deactivate.sh [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+fi
+unset -f unsetup
+if [[ "$CONDA_EUPS_BACKUP_unsetup" ]]; then
+  eval "$CONDA_EUPS_BACKUP_unsetup"
+  unset CONDA_EUPS_BACKUP_unsetup
+fi
+
+
+# restoring exisisting python path
+export PYTHONPATH=${CONDA_EUPS_BACKUP_PYTHONPATH}
```
Comment (beckermr) [inline recipes/eups/deactivate.sh:56]: If the python path was not set originally this will set it to nothing. I think you need a test if it is not empty.
